File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/TimeService.java
package com.github.arhor.simple.expense.tracker.service;

import java.time.LocalDate;
import java.time.ZonedDateTime;
import java.util.TimeZone;

import com.github.arhor.simple.expense.tracker.util.TemporalRange;

public interface TimeService {

    ZonedDateTime now();

    ZonedDateTime now(TimeZone timezone);

    TemporalRange<LocalDate> convertToDateRange(DateRangeCriteria criteria, TimeZone timezone);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/TimeService.kt
package com.github.arhor.simple.expense.tracker.service

import com.github.arhor.simple.expense.tracker.util.TemporalRange
import java.time.LocalDate
import java.time.ZonedDateTime
import java.util.TimeZone

interface TimeService {

    fun now(): ZonedDateTime

    fun now(timezone: TimeZone?): ZonedDateTime

    fun convertToDateRange(startDate: LocalDate?, endDate: LocalDate?, timezone: TimeZone?): TemporalRange<LocalDate>
}
